src/main/java/Entities/NormalGroup.java

```diff
@@ -69,5 +69,12 @@ public boolean setGroupLeader(String username) {
         return true;
     }
 
+    public void addMember(User user) {
+        groupMembers.put(user.getUsername(), user.getUsername());
+    }
+
+    public void removeFromRequests(User user) {
```

Comment:
I see that you're passing around a User object as a parameter. Just confirming -- this is allowed since your interactor will be the only thing calling this method and so it is able to directly interact with the entities and pass them around? When I added my use case methods to the entity classes, I had groupname and username as parameters, not the object itself.
If either way is fine, or if you think passing the entity objects is better, I'll change my methods so it's consistent with yours.

Reply:
Yes, it doesn't violate clean architecture since the method is called in the interactor. However, I noticed that the memberRequests hashmap was changed to map username to username(?) instead of username to user (we might want to consider changing it into an arraylist then) so I think it's better to just pass the username as a parameter so we can add it directly. I'll change mine to match yours.